modules/core/src/main/java/org/apache/synapse/deployers/ProxyServiceDeployer.java

```diff
@@ -159,6 +159,7 @@ public String updateSynapseArtifact(OMElement artifactConfig, String fileName,
                 }
                 ProxyService currentProxy = getSynapseConfiguration().getProxyService(existingArtifactName);
                 currentProxy.stop(getSynapseConfiguration());
+                currentProxy.destroy();
```

Comment:
should we destroy at line 118 also ?

Reply:
and this seems not destroyed properly in APIDeployer undeploySynapseArtifact method too , shall we fix that too with this.

Reply:
+1, fixed both in d17210e81d6224de123b303528a9e1cca5da0c1b